### PR TITLE
[TFLite] 16x8 reference kernel for PAD operator

### PR DESCRIPTION
**Pull request by wwwind**
In this PR I extend the operator PAD for 16x8 quantization mode.
This operator is present in Mobilenet V1/V2 from keras.applications.

### CODE REVIEW

**Comment by yyoon** [comment]
cc/ @renjie-liu.

File: tensorflow/lite/kernels/pad.cc [hunk 1]
```diff
@@ -208,29 +244,10 @@ TfLiteStatus Eval(TfLiteContext* context, TfLiteNode* node) {
       }
     } break;
     case kTfLiteInt8: {
```
Comment (renjie-liu) [inline tensorflow/lite/kernels/pad.cc:246]: shall we change the uint8 path as well?

File: tensorflow/lite/kernels/internal/reference/pad.h [hunk 2]
```diff
@@ -168,6 +168,16 @@ inline void PadImageStyle(const tflite::PadParams& op_params,
       output_data);
 }
 
+template <typename P>
+inline void PadImageStyle(const tflite::PadParams& op_params,
```
Comment (renjie-liu) [inline tensorflow/lite/kernels/internal/reference/pad.h:172]: shall we refactor a little bit that input & output types can be template arguments?

**Comment by wwwind** [comment]
Hi @renjie-liu Thanks for the review. I addressed your comments - please take a look.

**Review by renjie-liu — DISMISSED** [review]

**Review by rthadur — DISMISSED** [review]

**Comment by rthadur** [comment]
@yyoon @renjie-liu this PR has been failing internally , can you please help with this ?

File: tensorflow/lite/kernels/pad.cc [hunk 3]
```diff
@@ -119,6 +119,42 @@ TfLiteStatus Prepare(TfLiteContext* context, TfLiteNode* node) {
...
+        GetTensorShape(op_context.output),
+        GetTensorData<integer_type>(op_context.output));
+  } else {
+    optimized_ops::Pad(op_params, GetTensorShape(op_context.input),
+                       GetTensorData<integer_type>(op_context.input),
+                       &pad_value_copy, GetTensorShape(op_context.output),
+                       GetTensorData<integer_type>(op_context.output));
+  }
+}
+
```
Comment (renjie-liu) [inline tensorflow/lite/kernels/pad.cc:157]: this will require a return

**Review by renjie-liu — DISMISSED** [review]

**Review by rthadur — DISMISSED** [review]

**Comment by gbaned** [comment]
@wwwind Can you please resolve conflicts? Thanks!

**Review by rthadur — DISMISSED** [review]

**Comment by rthadur** [comment]
@renjie-liu  @yyoon can you please help with internal failures ?

**Comment by yyoon** [comment]
Hmm. AFAICT, the error seems unrelated to this PR. Let me try re-running the internal CI builds.

**Comment by yyoon** [comment]
@rthadur All checks have passed now.

File: tensorflow/lite/toco/tflite/op_version.cc [hunk 4]
```diff
@@ -104,10 +104,12 @@ string GetMinimumRuntimeVersionForModel(const Model& model) {
           {{OperatorType::kMul, 4}, kPendingReleaseOpVersion},
           {{OperatorType::kPad, 1}, "1.5.0"},
           {{OperatorType::kPad, 2}, "1.14.0"},
+          {{OperatorType::kPad, 3}, kPendingReleaseOpVersion},
```
Comment (renjie-liu) [inline tensorflow/lite/toco/tflite/op_version.cc:107]: we should modify tensorflow/lite/tools/versioning/runtime_version.cc

**Comment by wwwind** [comment]
Hi @renjie-liu Corrected. Please take a look. Thanks!

**Review by rthadur — APPROVED** [review]

**Review by jdduke — APPROVED** [review]

**Pull request merged**